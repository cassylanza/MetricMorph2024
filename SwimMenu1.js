import React from 'react'
import { TouchableOpacity, Text, View, StyleSheet, Image } from 'react-native'
import { Actions } from 'react-native-router-flux'

const SwimMenu1 = () => {

  const goToTrackMenu = () => {
    Actions.trackmenu1()
  }

  const goToSwimMenu = () => {
    Actions.swimmenu1()
  };

  const goToConvertMeterYardEntry = () => {
    Actions.convmeteryardinfoentry()
  }

  const goToHundredProjectionInfoEntry = () => {
    Actions.hundredinfoentry()
  }

  const goToRaceProjectionInfoEntry = () => {
    Actions.raceinfoentry()
  }

  return (
    <View style = {styles.containerScreen}>
      <Image style = {styles.logo} source={require('./MetricMorphLogo.png')} />
      <View style = {styles.swimMenuButton}>
        <TouchableOpacity onPress = {goToConvertMeterYardEntry}>

         <Text style = {styles.textSwimMenu}>Meter/yard conversion</Text>

        </TouchableOpacity>
</View>
<View style = {styles.swimMenuButton}>
        <TouchableOpacity onPress = {goToHundredProjectionInfoEntry}>
         <Text style = {styles.textSwimMenu}>6x100 Projection</Text>
        </TouchableOpacity>
</View>
<View style = {styles.swimMenuButton}>
        <TouchableOpacity onPress = {goToRaceProjectionInfoEntry}>

         <Text style = {styles.textSwimMenu}>Race Projection</Text>

        </TouchableOpacity>
        </View>

       <View style = {styles.containerTaskBar}>
          <TouchableOpacity onPress = {goToSwimMenu}>
            <Text style = {styles.textTaskBarSwim}> Swim Menu </Text>
          </TouchableOpacity>
          <TouchableOpacity onPress = {goToTrackMenu}>
            <Text style = {styles.textTaskBar}> Track Menu </Text>
          </TouchableOpacity>
        </View>
      </View>

  )
}
export default SwimMenu1

const styles = StyleSheet.create ({
   containerScreen: {
     flexDirection: 'column',
     alignItems: 'center',
     backgroundColor: 'white',
    flex: 1
   },

   logo: {
      marginTop: 80,
      marginBottom: 75
   },

   swimMenuButton: {
  padding: 10,
  margin: 20,
  width: 280,
  height: 60,
  borderRadius: 25,
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'center',
  backgroundColor: 'green'

   },

  containerTaskBar: {
     flex: 1,
     flexDirection: 'row',
     position: 'absolute',
     alignItems: 'center',
     height: '12%',
     width: '100%',
     bottom: 0,
     left: 0,
     borderWidth: 1,
     backgroundColor: 'white'
  },

   textTaskBar: {
     width: 188,
     height: 100,
     textAlign: 'center',
     padding: 37,
     fontSize: 20,


   },
   textTaskBarSwim: {
     width: 188,
     height: 100,
     textAlign: 'center',
     padding: 37,
     fontSize: 20,
     fontWeight: 'bold',
     color: 'blue'
   },
   textSwimMenu: {
      textAlign: 'center',
      fontSize: 20,
      color: 'white'
   }
})
