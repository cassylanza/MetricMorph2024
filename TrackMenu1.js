import React from 'react'
import { TouchableOpacity, Text, View, StyleSheet, Image } from 'react-native'
import { Actions } from 'react-native-router-flux'
import { FontAwesomeIcon }  from '@fortawesome/react-native-fontawesome'
import { faPersonSwimming } from '@fortawesome/free-solid-svg-icons/faPersonSwimming'
import { faPersonRunning } from '@fortawesome/free-solid-svg-icons/faPersonRunning'


const TrackMenu1 = () => {

  // navigation constants
  const goToSwimMenu = () => {
    Actions.swimmenu1()
  }

  const goToConvertInfoEntry = () => {
    Actions.convperfinfoentry()
  }

  const goToSplitsInfoEntry = () => {
    Actions.splitsinfoentry()
  }

  const goToEffortInfoEntry = () => {
    Actions.effortinfoentry()
  }

  return (
    <View style = {styles.containerScreen}>
    {/*  <Image style = {styles.logo} source={require('./MetricMorphLogo.png')} /> */}
      <View style = {{padding: '50%'}}>
        <TouchableOpacity onPress = {goToConvertInfoEntry} style = {styles.trackMenuButton}>

           <Text style = {styles.textTrackMenu}>Convert Race</Text>

        </TouchableOpacity>

        <TouchableOpacity onPress = {goToSplitsInfoEntry} style = {styles.trackMenuButton}>

           <Text style = {styles.textTrackMenu}>Splits</Text>

          </TouchableOpacity>

        <TouchableOpacity onPress = {goToEffortInfoEntry} style = {styles.trackMenuButton}>

           <Text style = {styles.textTrackMenu}>% Effort</Text>

          </TouchableOpacity>
      </View>

      <View style = {styles.containerTaskBar}>
        <TouchableOpacity onPress = {goToSwimMenu} style={{alignItems: 'center', marginBottom: '2%'}}>
            <FontAwesomeIcon
              icon={faPersonSwimming}
              size={30}
              color='gray'
              style = {{marginBottom: 2}}
            />
            <Text style = {{color: 'gray'}}>Swimming</Text>
        </TouchableOpacity>
        <View style = {{alignItems: 'center', marginBottom: '2%'}}>
          <FontAwesomeIcon
            icon={faPersonRunning}
            size={30}
            color='#00b3b3'
            style = {{marginBottom: 2}}
          />
          <Text style = {{color: '#00b3b3'}}>Track</Text>
        </View>
      </View>
    </View>
  )
}
export default TrackMenu1

const styles = StyleSheet.create ({
   containerScreen: {
     flexDirection: 'column',
     alignItems: 'center',
     backgroundColor: 'white',
     flex: 1
   },
   logo: {
      marginTop: 80,
      marginBottom: 75
   },
   trackMenuButton: {
     padding: 10,
     margin: 20,
     width: 280,
     height: 60,
     borderRadius: 25,
     alignItems: 'center',
     justifyContent: 'center',
     flexDirection: 'center',
     backgroundColor: '#00b3b3'
   },
   textTrackMenu: {
      textAlign: 'center',
      fontSize: 20,
      color: 'white'
   },
   containerTrackMenu: {
     textAlign: 'center',
     flexDirection: 'column',
     justifyContent: 'center',
     alignItems: 'center',
     backgroundColor: 'white',
     height: 420,
     width: 300
   },
   containerTaskBar: {
     flex: 1,
     flexDirection: 'row',
     position: 'absolute',
     justifyContent: 'space-around',
     alignItems: 'center',
     height: '12%',
     width: '100%',
     bottom: 0,
     left: 0,
     borderWidth: 1,
     borderTopColor: 'lightgray',
     backgroundColor: 'white'
   },
   textTaskBar: {
     width: 188,
     height: 100,
     textAlign: 'center',
     padding: 37,
     fontSize: 20,
   }
})
