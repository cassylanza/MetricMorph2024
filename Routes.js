import React from 'react'
import { Router, Scene } from 'react-native-router-flux'
import SwimMenu1 from './SwimMenu1.js'
import TrackMenu1 from './TrackMenu1.js'
import ConvertPerfInfoEntry from './ConvertPerfInfoEntry.js'
import SplitsInfoEntry from './SplitsInfoEntry.js'
import EffortInfoEntry from './EffortInfoEntry.js'
import ConvertMeterYardInfoEntry from './ConvertMeterYardInfoEntry.js'
import HundredProjectionInfoEntry from './HundredProjectionInfoEntry.js'
import RaceProjectionInfoEntry from './RaceProjectionInfoEntry.js'




const Routes = () => (
  <Router>
    <Scene key = "root">
      <Scene key = "swimmenu1" component = {SwimMenu1} title = "SWIM" initial = {true} />
      <Scene key = "trackmenu1" component = {TrackMenu1} title = "TRACK" />
      <Scene key = "convperfinfoentry" component = {ConvertPerfInfoEntry} title = "CONVERT RACE" />
      <Scene key = "splitsinfoentry" component = {SplitsInfoEntry} title = "SPLITS INFO ENTRY" />
      <Scene key = "effortinfoentry" component = {EffortInfoEntry} title = "EFFORT INFO ENTRY"/>
      <Scene key = "convmeteryardinfoentry" component = {ConvertMeterYardInfoEntry} title = "CONVERT METERS/YARDS" />
      <Scene key = "hundredinfoentry" component = {HundredProjectionInfoEntry} title = "6x100 PROJECTION" />
      <Scene key = "raceinfoentry" component = {RaceProjectionInfoEntry} title = "RACE BASED PROJECTION" />
    </Scene>
  </Router>
)

export default Routes
