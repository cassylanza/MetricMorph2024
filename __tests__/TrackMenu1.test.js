import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import { Actions } from 'react-native-router-flux'; // Mocking react-native-router-flux
import '@testing-library/jest-dom';
import Routes from '../Routes'; // Importing the component to test
import SwimMenu1 from '../SwimMenu1'; // Importing SwimMenu1 component
import TrackMenu1 from '../TrackMenu1'; // Importing TrackMenu1 component
import { FontAwesomeIcon }  from '@fortawesome/react-native-fontawesome'
import { faPersonSwimming } from '@fortawesome/free-solid-svg-icons/faPersonSwimming'
import { faPersonRunning } from '@fortawesome/free-solid-svg-icons/faPersonRunning'

// Mocking react-native-router-flux actions
jest.mock('react-native-router-flux', () => ({
  Actions: {
    swimmenu1: jest.fn(),
    trackmenu1: jest.fn(),
    convperfinfoentry: jest.fn(),
    conversionresults: jest.fn(),
    splitsinfoentry: jest.fn(),
    splitsresults: jest.fn(),
    effortinfoentry: jest.fn(),
    effortresults: jest.fn(),
  },
}));

describe('TrackMenu1', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TrackMenu1 />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('first menu option appears', () => {
    const { getByText } = render(<TrackMenu1 />);
    expect(getByText(" CONVERT RACE ")).toBeTruthy();
  });

  it('second menu option appears', () => {
    const { getByText } = render(<TrackMenu1 />);
    expect(getByText(" SPLITS ")).toBeTruthy();
  });

  it('third menu option appears', () => {
    const { getByText } = render(<TrackMenu1 />);
    expect(getByText(" % EFFORT ")).toBeTruthy();
  });

  it('task bar swim button appears', () => {
    const { getByText } = render(<TrackMenu1 />);
    expect(getByText("Swim Menu")).toBeTruthy();
  });

  it('task bar track button appears', () => {
    const { getByText } = render(<TrackMenu1 />);
    expect(getByText("Track Menu")).toBeTruthy();
  });

  test('shows conversion info entry when Convert Race is pressed', () => {
    render(
      <NavigationContainer>
        <TrackMenu1 />
      </NavigationContainer>
    )

    fireEvent.press(screen.getByText(' CONVERT RACE '));

    expect(screen.getByText(' CONVERT RACE ')).toBeTruthy();
  });

  test('shows splits info entry when Splits is pressed', () => {
    render(
      <NavigationContainer>
        <TrackMenu1 />
      </NavigationContainer>
    )

    fireEvent.press(screen.getByText(' SPLITS '));

    expect(screen.getByText(' SPLITS ')).toBeTruthy();
  });

  test('shows % effort info entry when % EFFORT is pressed', () => {
    render(
      <NavigationContainer>
        <TrackMenu1 />
      </NavigationContainer>
    )

    fireEvent.press(screen.getByText('% Effort'));

    expect(screen.getByText('Effort')).toBeTruthy();
  });

  test('show swim menu when swim menu button pressed', () => {
    render(
      <NavigationContainer>
        <TrackMenu1 />
      </NavigationContainer>
    )

    fireEvent.press(screen.getByText('Swim Menu'));

    expect(screen.getByText('Swim Menu')).toBeTruthy();
  })

  describe('TrackMenu1', () => {
  it('renders swim icon', () => {
    const { getByA11yRole } = render(<TrackMenu1 />);
    const swimIcon = getByA11yRole('image');
    expect(swimIcon.props.style[1].color).toBe('gray');
  });

  it('renders track icon', () => {
    const { getByA11yRole } = render(<TrackMenu1 />);
    const trackIcon = getByA11yRole('image');
    expect(trackIcon.props.style[1].color).toBe('#00b3b3');
  });
});


})
