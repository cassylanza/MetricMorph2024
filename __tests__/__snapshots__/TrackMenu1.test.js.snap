// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackMenu1 renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../MetricMorphLogo.png",
      }
    }
    style={
      {
        "marginBottom": 75,
        "marginTop": 80,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 25,
        "flexDirection": "center",
        "height": 60,
        "justifyContent": "center",
        "margin": 20,
        "padding": 10,
        "width": 280,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        CONVERT RACE
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 25,
        "flexDirection": "center",
        "height": 60,
        "justifyContent": "center",
        "margin": 20,
        "padding": 10,
        "width": 280,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        SPLITS
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 25,
        "flexDirection": "center",
        "height": 60,
        "justifyContent": "center",
        "margin": 20,
        "padding": 10,
        "width": 280,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        EFFORT
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderWidth": 1,
        "bottom": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": "12%",
        "left": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "height": 100,
            "padding": 37,
            "textAlign": "center",
            "width": 188,
          }
        }
      >
         Swim Menu 
      </Text>
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "height": 100,
          "padding": 37,
          "textAlign": "center",
          "width": 188,
        }
      }
    >
      <Text
        style={
          {
            "color": "blue",
            "fontWeight": "bold",
          }
        }
      >
         Track Menu 
      </Text>
    </Text>
  </View>
</View>
`;
